### PR TITLE
[C++ Worker]Update cpp worker doc

### PR DESCRIPTION
**Pull request by qicosmos**
## Why are these changes needed?

Update c++ worker document, add some placement group examples.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests

### CODE REVIEW

File: doc/source/placement-group.rst [hunk 1]
```diff
@@ -237,6 +267,56 @@ Let's create a placement group. Recall that each bundle is a collection of resou
...
+
+        class Counter {
+        public:
+          std::string Ping() {
+            return "pong";
+          }
+        };
+
+        // Factory function of Counter class.
+        static Counter *CreateCounter() {
```
Comment (SongGuyang) [inline doc/source/placement-group.rst:286]: Better to move this static factory function out.

File: doc/source/placement-group.rst [hunk 2]
```diff
@@ -237,6 +267,56 @@ Let's create a placement group. Recall that each bundle is a collection of resou
...
+        };
+
+        RAY_REMOTE(&Counter::Ping, CreateCounter);
+
+        // Construct a list of bundles.
+        std::vector<std::unordered_map<std::string, double>> bundles{{{"CPU", 2.0}}};
+
+        // Create a placement group and make sure its creation is successful.
+        ray::internal::PlacementGroupCreationOptions options{
+            false, name, bundles, ray::internal::PlacementStrategy::STRICT_SPREAD};
```
Comment (SongGuyang) [inline doc/source/placement-group.rst:297]:
```suggestion
            false, "my_pg", bundles, ray::PlacementStrategy::STRICT_SPREAD};
```

File: doc/source/placement-group.rst [hunk 3]
```diff
@@ -284,6 +364,25 @@ Let's create a placement group. Recall that each bundle is a collection of resou
...
+
+        std::vector<std::unordered_map<std::string, double>> bundles{{{"GPU", 2.0}, {"extra_resource", 2.0}}};
+
+        /**
+         * Reserve bundles with strict pack strategy.
+         * It means Ray will reserve 2 "GPU" and 2 "extra_resource" on the same node (strict pack) within a Ray cluster.
+         * Using this placement group for scheduling actors or tasks will guarantee that they will
+         * be colocated on the same node.
+         */
+        ray::internal::PlacementGroupCreationOptions options{
```
Comment (SongGuyang) [inline doc/source/placement-group.rst:379]:
```suggestion
        ray::PlacementGroupCreationOptions options{
```

File: doc/source/placement-group.rst [hunk 4]
```diff
@@ -284,6 +364,25 @@ Let's create a placement group. Recall that each bundle is a collection of resou
...
+            false, name, bundles, ray::internal::PlacementStrategy::STRICT_PACK};
```
Comment (SongGuyang) [inline doc/source/placement-group.rst:380]:
```suggestion
            false, "my_pg", bundles, ray::PlacementStrategy::STRICT_PACK};
```

File: doc/source/placement-group.rst [hunk 5]
```diff
@@ -354,6 +453,44 @@ Now let's define an actor that uses GPU. We'll also define a task that use ``ext
...
 
+  .. group-tab:: C++
+
+    .. code-block:: c++
+
+      class Counter {
+      public:
+        Counter(int init_value) : value(init_value){}
+        int GetValue() {return value;}
+        std::string Ping() {
```
Comment (SongGuyang) [inline doc/source/placement-group.rst:464]: `Ping` should be a static function? Move this function out from `Counter `?

File: doc/source/placement-group.rst [hunk 6]
```diff
@@ -112,6 +128,20 @@ Placement groups are atomically created - meaning that if there exists a bundle
...
+  .. group-tab:: C++
+
+    .. code-block:: c++
+
+      // Wait for the placement group to be ready within the specified time(unit is seconds).
+      boll ready = pg.Wait(60);
+      assert(ready);
+
+      // You can look at placement group states using this API.
+      std::vector<ray::PlacementGroup> all_placementGroup = ray::GetAllPlacementGroups();
```
Comment (raulchen) [inline doc/source/placement-group.rst:140]: `all_placement_group`

File: doc/source/placement-group.rst [hunk 7]
```diff
@@ -284,6 +364,25 @@ Let's create a placement group. Recall that each bundle is a collection of resou
...
         Assert.assertTrue(isCreated);
 
+    .. group-tab:: C++
+
+      .. code-block:: c++
+
+        std::vector<std::unordered_map<std::string, double>> bundles{{{"GPU", 2.0}, {"extra_resource", 2.0}}};
+
+        /**
+         * Reserve bundles with strict pack strategy.
```
Comment (raulchen) [inline doc/source/placement-group.rst:374]: use `//` for comments

**Review by raulchen — APPROVED** [review]

**Pull request merged**